clients/cobol-lsp-vscode-extension/src/__tests__/commands/RunAnalysisCLITest.spec.ts

```diff
@@ -0,0 +1,265 @@
+    registerCompletionItemProvider: jest.fn(),
+  },
+  window: {
+    activeTextEditor: {
+      document: {
+        uri: {
+          path: "/storagePath",
+          fsPath: "/storagePath",
+          scheme: "file",
+        },
+        getText: jest.fn(),
+      },
+    },
+    setStatusBarMessage: jest
+      .fn()
+      .mockImplementation(
+        async (_text: string, _hideWhenDone: Thenable<any>) => {
+          return Promise.resolve(true);
+        },
+      ),
+    showErrorMessage: jest.fn().mockReturnValue("test"),
+    showInformationMessage: jest
+      .fn()
+      .mockImplementation((message: string) => Promise.resolve(message)),
+    showQuickPick: jest.fn(),
+    onDidChangeActiveTextEditor: jest.fn(),
+    createQuickPick: jest
+      .fn()
+      .mockReturnValue({ onDidChangeSelection: jest.fn(), show: jest.fn() }),
+    createOutputChannel: jest.fn().mockReturnValue({
+      appendLine: jest.fn(),
+    }),
+    terminals: {
+      find: jest.fn(),
+    },
+    createTerminal: jest.fn().mockReturnValue({
+      sendText: jest.fn(),
+      show: jest.fn(),
+    }),
+  },
+  workspace: {
+    getConfiguration: jest.fn().mockReturnValue({
+      get: jest.fn().mockReturnValue(9),
+    }),
+    getWorkspaceFolder: jest.fn().mockReturnValue({ name: "workspaceFolder1" }),
+    onDidChangeConfiguration: jest
+      .fn()
+      .mockReturnValue("onDidChangeConfiguration"),
+    workspaceFolders: [{ uri: { fsPath: "ws-path" } } as any],
+    fs: {
+      createDirectory: jest.fn(),
+      stat: jest.fn().mockReturnValue(2),
+    },
+  },
+  Uri: {
+    file: jest.fn().mockReturnValue("workspaceFolder2"),
+    parse: jest.fn().mockReturnValue({ path: "/storagePath" }),
+  },
+}));
+
+const context: any = {
+  extensionUri: { fsPath: "/test" },
+  globalStorageUri: { fsPath: "/storagePath" },
+  subscriptions: [],
+};
+
+describe("Test Analysis CLI command functionality", () => {
+  test("Cobol Analysis - Java", async () => {
+    const testAnalysis = new RunAnalysis(
+      vscode.Uri.parse(context.globalStorageUri),
+      vscode.Uri.parse(context.extensionUri),
+    );
+    vscode.window.showQuickPick = jest.fn().mockReturnValue("Java");
+
+    const runCobolAnalysisCommandSpy = jest.spyOn(
+      testAnalysis as any,
+      "runCobolAnalysisCommand",
+    );
+    const getCurrentFileLocationSpy = jest.spyOn(
+      testAnalysis as any,
+      "getCurrentFileLocation",
+    );
+    const buildJavaCommandSpy = jest.spyOn(
+      testAnalysis as any,
+      "buildJavaCommand",
+    );
+    const buildNativeCommandSpy = jest.spyOn(
+      testAnalysis as any,
+      "buildNativeCommand",
+    );
+    const buildAnalysisCommandPortionSpy = jest.spyOn(
+      testAnalysis as any,
+      "buildAnalysisCommandPortion",
+    );
+    const sendToTerminalSpy = jest.spyOn(testAnalysis as any, "sendToTerminal");
+
+    await testAnalysis.runCobolAnalysisCommand();
+
+    expect(runCobolAnalysisCommandSpy).toHaveBeenCalled();
```

Comment:
Please add checks that the routines are called with expected parameters or that they returned expected result.

Reply:
Resolved in latest commit.